src/lib/decoders.ts

```diff
@@ -0,0 +1,80 @@
+import BigNumber from 'bignumber.js';
+import { defaultAbiCoder } from 'ethers/lib/utils';
+import { parseInt, result } from 'lodash';
+import { BN_0 } from '../bignumber-constants';
+import { MultiResult } from './multi-wrapper';
+
+export const uintDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return BigInt(defaultAbiCoder.decode(['uint'], result.returnData)[0]);
+};
+
+export const uint256ArrayDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return BigInt(defaultAbiCoder.decode(['uint256[]'], result.returnData)[0]);
+};
+
+export const uin256DecodeToBigNumber = (
+  result: MultiResult<string>,
+): BigNumber => {
+  if (!result.success) {
+    return BN_0;
+  }
+  return new BigNumber(
+    defaultAbiCoder.decode(['uint256'], result.returnData)[0],
+  );
+};
+
+export const uin256DecodeToNumber = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseInt(
+    defaultAbiCoder.decode(['uint256'], result.returnData)[0],
+    10,
+  );
+};
+
+export const uin256DecodeToFloat = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseFloat(defaultAbiCoder.decode(['uint256'], result.returnData)[0]);
+};
+
+export const uin128DecodeToFloat = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseFloat(defaultAbiCoder.decode(['uint128'], result.returnData)[0]);
+};
```

Comment:
@ColonelJ :
https://github.com/paraswap/paraswap-dex-lib/pull/166#discussion_r955979981

Reply:
This come from curveV1 code from api there we were parsing float for some reason